src/DynamoCoreWpf/ViewModels/Search/SearchViewModel.cs

```diff
@@ -992,13 +993,17 @@ internal IEnumerable<NodeSearchElementViewModel> Search(string search, bool useL
         /// </summary>
         /// <param name="nodeName">Name of the node</param>
         /// <param name="nodeCategory">Full Category of the node</param>
+        /// <param name="parameters">Node input parameters</param>
         /// <returns></returns>
-        private NodeSearchElementViewModel FindViewModelForNodeNameAndCategory(string nodeName, string nodeCategory)
+        private NodeSearchElementViewModel FindViewModelForNodeNameAndCategory(string nodeName, string nodeCategory, string parameters)
         {
             var result = Model.SearchEntries.Where(e => {
                 if (e.Name.Equals(nodeName) && e.FullCategoryName.Equals(nodeCategory))
                 {
-                    return true;
+                    if (e.Parameters == null)
+                        return string.IsNullOrEmpty(parameters);
+                    else
+                        return e.Parameters.Equals(parameters);
```

Comment:
Can you put comments for this block, not too intuitive to me

Reply:
I've added more comments please let me know if the explanation is clear. Thanks
commit: https://github.com/DynamoDS/Dynamo/pull/14136/commits/6bc39b724b1290d5c4bdd61cc3cf6a035f744d97